### PR TITLE
Configure GCB triggers with Terraform

### PR DESCRIPTION
**Pull request by will-cromar**
Also support platform tagging so the names don't conflict.

We can use Terraform to manage our GCB build triggers declaratively, instead of copying and pasting fields across Cloud Console:

e.g.

```
module "nightly-py37-tpuvm" {
  source = "./modules/trigger"

  release = "nightly"
  python_version = "3.7"
  platform = "tpuvm"
  docker_build_args = [ "tpuvm=1" ]
}

module "nightly-py38-tpuvm" {
  source = "./modules/trigger"

  release = "nightly"
  python_version = "3.8"
  platform = "tpuvm"
  docker_build_args = [ "tpuvm=1" ]
}

module "nightly-py38-tpunode" {
  source = "./modules/trigger"

  release = "nightly"
  python_version = "3.8"
  platform = "tpunode"
  docker_build_args = [ "tpuvm=0" ]
}

module "nightly-py38-cuda112" {
  source = "./modules/trigger"

  release = "nightly"
  python_version = "3.8"
  platform = "cuda112"
  docker_build_args = [ "tpuvm=0,cuda=1"]
}
```

### CODE REVIEW

**Comment by will-cromar** [comment]
The scheduled triggers for nightly are currently running (see checks on this PR). Most of them are passing, except for the GPU build, which appears to be flaking due to lack of resources. 

This still needs some follow up before it's ready for use full-time:

- Learn the best practice for sharing Terraform state with the team so we can run it outside of my workstation.
- Add a [GCB custom pool](https://cloud.google.com/build/docs/private-pools/private-pools-overview) to the config with more resources.
- Configure a more restricted service account to run in Cloud Scheduler.
- Support release builds.

File: docker/experimental/terraform/main.tf [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  format        = "DOCKER"
+}
+
+module "nightly-py37-tpuvm" {
+  source = "./modules/trigger"
+
+  release = "nightly"
+  python_version = "3.7"
+  platform = "tpuvm"
+  docker_build_args = [ "tpuvm=1" ]
```
Comment (steventk-g) [inline docker/experimental/terraform/main.tf:25]: Does this mean that these images will build with CPP tests? Any concerns there?

File: docker/experimental/terraform/modules/trigger/main.tf [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    _BUILD_ARGS = join(",", var.docker_build_args)
+    _PYTHON_VERSION = var.python_version
+  }
+}
+
+resource "google_cloud_scheduler_job" "trigger-schedule" {
+  name = format("%s-schedule", local.trigger_name)
+  region = "us-central1"
+
+  schedule = "0 0 * * *"
```
Comment (steventk-g) [inline docker/experimental/terraform/modules/trigger/main.tf:53]: Could you add a comment explaining what the schedule string means?

File: docker/experimental/terraform/modules/trigger/main.tf [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  time_zone = "America/Los_Angeles"
+
+  http_target {
+    http_method = "POST"
+    uri = "https://cloudbuild.googleapis.com/v1/projects/${google_cloudbuild_trigger.build-trigger.project}/triggers/${google_cloudbuild_trigger.build-trigger.trigger_id}:run"
+
+    oauth_token {
+      # TODO: Use a better service account
+      service_account_email = "164006649440-compute@developer.gserviceaccount.com"
```
Comment (steventk-g) [inline docker/experimental/terraform/modules/trigger/main.tf:62]: Is there any risk to exposing this email in a public repo via this PR?

File: docker/experimental/terraform/main.tf (hunk 1 above)
Comment (will-cromar) [reply docker/experimental/terraform/main.tf:25]: The default is still to build without the CPP tests (see the top of the Dockerfile) in this PR.

File: docker/experimental/terraform/modules/trigger/main.tf [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+resource "google_cloudbuild_trigger" "build-trigger" {
+  location = "global"
+  name = local.trigger_name
+
+  source_to_build {
+    uri = "https://github.com/pytorch/xla"
+    repo_type = "GITHUB"
+    # TODO: make branch configurable
+    ref = "refs/heads/master"
```
[the remark this answers is not in the document]
Comment (will-cromar) [reply docker/experimental/terraform/modules/trigger/main.tf:31]: I like this idea. Every time the CI passes we could update a "green" branch to that commit plus a pin to the upstream PyTorch commit that the build passed with. Happy to update the branch here if someone has time to implement that.

File: docker/experimental/terraform/modules/trigger/main.tf (hunk 2 above)
Comment (will-cromar) [reply docker/experimental/terraform/modules/trigger/main.tf:54]: Linked to Cloud Scheduler's documentation on cron schedule format.

File: docker/experimental/terraform/modules/trigger/main.tf (hunk 3 above)
Comment (will-cromar) [reply docker/experimental/terraform/modules/trigger/main.tf]: Not that I know of. The project number is functionally equivalent to our GCP project name AFAIK (which you can already find easily in this repository).

Changed this to a different SA in our project that was already created for this purpose (triggering builds from Cloud Scheduler) but wasn't being used.

**Pull request merged**